### PR TITLE
v8.1.0

### PR DESCRIPTION
**Pull request by rmarinleal**

### CODE REVIEW

File: controllers/front/payment.php [hunk 1]
```diff
@@ -200,6 +206,7 @@ public function postProcess()
             $orderConfiguration
                 ->setChannel($orderChannel)
                 ->setUrls($orderConfigurationUrls)
+                // ->setPurchaseCountry(substr(Mage::app()->getLocale()->getLocaleCode(), -2, 2))
```
Comment (pgarcess) [inline controllers/front/payment.php]: Commented code

File: controllers/front/payment.php [hunk 2]
```diff
@@ -154,31 +159,32 @@ public function postProcess()
...
                 $product
-                    ->setAmount((int) (100 * $item['price_wt']))
+                    ->setAmount((string) floor(100 * $item['price_wt']))
                     ->setQuantity($item['quantity'])
                     ->setDescription($item['name']);
                 $details->addProduct($product);
             }
 
             $orderShoppingCart = new \Pagantis\OrdersApiClient\Model\Order\ShoppingCart();
+            $totalAmount = (string) floor(100 * $cart->getOrderTotal(true));
```
Comment (pgarcess) [inline controllers/front/payment.php]: total_amount: REQUIRED Int

File: controllers/front/payment.php [hunk 3]
```diff
@@ -127,7 +130,8 @@ public function postProcess()
                 ->setCountryCode('ES')
                 ->setCity($billingAddress->city)
                 ->setAddress($billingAddress->address1 . ' ' . $billingAddress->address2)
-                ->setDni($billingAddress->dni)
```
Comment (pgarcess) [inline controllers/front/payment.php]: Taking a look to last version, the code contains the line: ->setDni($billingAddress->dni). I have been trying to find the variable DNI in the methods getNationalId and I did not find it.

File: views/templates/hook/checkout.tpl [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+        <p class="payment_module Pagantis ps_version_{$ps_version|escape:'quotes'}">
+            <a class="pagantis-checkout ps_version_{$ps_version|escape:'quotes'} locale_{$locale|escape:'quotes'}" href="{$paymentUrl|escape:'html'}" title="{$pagantisTitle|escape:'quotes'}">
+                {$pagantisTitle|escape:'quotes'}
+                <span class="PagantisSimulator ps_version_{$ps_version|escape:'quotes'}"></span>
+            </a>
+        </p>
+        <script type="text/javascript">
+            function checkSimulatorContent() {
+                var pgContainer = document.getElementsByClassName("PagantisSimulator");
+                if(typeof pgContainer != 'undefined' && typeof pgContainer[0] != 'undefined') {
```
Comment (pgarcess) [inline views/templates/hook/checkout.tpl]: This line could be replaced by: "pgContainer.length > 0"

File: views/templates/hook/product-simulator.tpl [hunk 5]
```diff
@@ -7,9 +7,38 @@
  *}
 {if ($pagantisIsEnabled && $pagantisSimulatorIsEnabled)}
     <script>
+        function checkSimulatorContent() {
```
Comment (pgarcess) [inline views/templates/hook/product-simulator.tpl]: We need to set a limit of attemps, maybe we're using lang=it and simulator is not showing. Check if could help you this code: https://github.com/pagantis/woocommerce/blob/master/templates/product_simulator.php#L17

File: controllers/front/payment.php (hunk 1 above)
Comment (rmarinleal) [reply controllers/front/payment.php]: Done

File: controllers/front/payment.php (hunk 2 above)
Comment (rmarinleal) [reply controllers/front/payment.php]: Pasamos string en todos los módulos.
https://github.com/pagantis/magento-1X/blob/master/extension/app/code/community/Pagantis/Pagantis/controllers/PaymentController.php#L252

File: controllers/front/payment.php (hunk 3 above)
Comment (rmarinleal) [reply controllers/front/payment.php]: Added dni variable

File: views/templates/hook/checkout.tpl (hunk 4 above)
Comment (rmarinleal) [reply views/templates/hook/checkout.tpl]: Changed

File: views/templates/hook/product-simulator.tpl (hunk 5 above)
Comment (rmarinleal) [reply views/templates/hook/product-simulator.tpl]: Added attempts behavior

**Pull request merged**